codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/parse/JsonParserGenerator.kt

```diff
@@ -197,31 +183,9 @@ class JsonParserGenerator(
         )
     }
 
-    fun renderStructure(
-        writer: RustWriter,
-        structureShape: StructureShape,
-        includedMembers: List<MemberShape>,
-    ) {
+    override fun serverParser(structureShape: StructureShape, includedMembers: List<MemberShape>): RuntimeType? {
```

Comment:
you _may_ want to have this accept an `operationShape` instead (then the task of using the index to retrieve the members is identical to the `operationParser` code above.

The methods would be named something like:

`clientOperationParser` and `serverOperationParser` (or `operationInputParser`, `operationOutputParser`)

Not necessarily suggesting this in this diff, but we should probably try to be consistent eventually

Reply:
Working on it, I will also try to do something similar in the `JsonSerializerGenerator` to have a little more consistency.